py/tables/fcoll_zreion.py

```diff
@@ -0,0 +1,422 @@
+
+plt.xlabel('z', fontsize=20)
+plt.ylabel('F_coll', fontsize=20)
+plt.tick_params(labelsize=20)
+plt.legend(loc='best', fontsize=10)
+plt.yscale('log')
+plt.axhline(1, linestyle='--')
+plt.legend()
+plt.show()
+
+#red dotted line should match blue curve
+
+
+# compute the final conditional fcoll from equation 3 in overleeaf
+
+# In[42]:
+
+
+
+print(F_coll.shape)
+print(cond_fcollPS.shape)
+
+#F_coll_final = 
+
+F_coll = np.nan_to_num(F_coll)
+F_coll_PS = np.nan_to_num(F_coll_PS)
+cond_fcollPS[:,:,:] = np.nan_to_num(cond_fcollPS[:,:,:])
+
+F_coll_final = (F_coll * cond_fcollPS[:,:,:]) / F_coll_PS #eqn 3...F_coll_PS[i] = bar(fcollPS), = fcollPS
+print(F_coll_final)
+
+#print(F_coll_final.shape)
+#print(F_coll_PS[1,:,:])
+
+
+# plot the values to be sure they agree since we are begining with just PS for checks
+
+# In[18]:
+
+
+plt.plot(zvals, (F_coll_PS[:,0,0]), 'b-', label = 'unconditional from colosus') #
+
+
+plt.semilogy(zvals,F_coll_final[:,0,0], 'r-', label='Final conditional fcoll from equation')
+plt.xlabel('z', fontsize=20)
+plt.ylabel('F_coll', fontsize=20)
+    
+plt.plot(zvals,cond_fcollPS[:,-1,5], 'g:', label='conditional fcoll PS')
+plt.yscale('log')
+plt.legend()
+plt.show()
+
+#red curve should be equal to greeen
+
+
+# In[ ]:
+
+
+# populate the tables
+fcoll_table = F_coll_final
+#print(fcoll_table)
+
+
+# In[ ]:
+
+
+for i in range(len(Rvals)):
+    for j in range(len(delta_Rvals)):
+        cond_fcoll_zvals = cond_fcoll(zvals, delta_Rvals[j], Rvals[i])
+        zeta_times_fcoll_zvals = zeta * cond_fcoll_zvals
+        z_of_zeta_times_fcoll = interp1d(zeta_times_fcoll_zvals, zvals, fill_value="extrapolate")
+        zreion_table[i,j] = z_of_zeta_times_fcoll(1.0)
+zreion_table = zreion_table.astype(np.float32)
+
+
+# In[ ]:
+
+
+print(len(zvals))
+print(len(F_coll_final))
+print(len(Rvals))
+
+
+# In[ ]:
+
+
+zreion_table = np.nan_to_num(zreion_table)
+print(zreion_table[:,49])#all values of zreion at rvals index 2
+print(zreion_table.shape)
+
+
+# In[ ]:
+
+
+#plot z_reion 1) vs R at a few values of delta_R, and 2) vs delta_R at a few values of R
+plt.plot(Rvals, zreion_table[:,49], 'o')
+plt.yscale('log')
+plt.show( )
+
+
+# In[ ]:
+
+
+# write the tables in binaries with minimal headers easily readable
+# in C
+# fcoll table
+#wrote tables in .txt as well so I can see the output
+
+tablefile2='fcoll_Mmin'+str(Mmin)+'.txt'  #add the mass function model to the file name
+#tablefile2='fcoll_Mmin'+str(Mmin)+str(mass_function.models[model])+'.txt'
+f2 = open(tablefile2,'w')
+fcoll_table.tofile(f2, sep=" ", format="%f")
+np.savetxt(tablefile2, np.column_stack([zvals, fcoll_table[:,0,0]]), fmt='%.18g')#zvals, Rvals, delta_Rvals, fcoll_table
+
+tablefile='fcoll_Mmin'+str(Mmin)+'.tab'
+f = open(tablefile,'wb')
+zrange = [zmin, zmax]
+zrange = np.asarray(zrange).astype(np.float32)
+dim   = [nz]
+dim   = np.asarray(dim).astype(np.int32)
+zrange.tofile(f)
+dim.tofile(f)
+fcoll_table.tofile(f)
+
+
+# In[ ]:
+
```

Comment:
We can clean the comments/code up a little here, but I'm happy to do this once we have the PR process nailed!